rules/prefer-error-cause.js

```diff
@@ -0,0 +1,388 @@
+/* eslint-disable capitalized-comments */
+/* eslint-disable brace-style */
+'use strict';
+const esquery = require('esquery');
+const {isCommaToken} = require('eslint-utils');
+const _ = require('lodash');
+const {matches, methodCallSelector} = require('./selectors/index.js');
+const {appendArgument, replaceReferenceIdentifier} = require('./fix/index.js');
+const getReferences = require('./utils/get-references.js');
+
+const ERROR = 'error';
+const SUGGESTION = 'suggestion';
+
+const messages = {
+	[ERROR]: 'Prefer specifying an old error as `cause` option where rethrowing the error.',
+	[SUGGESTION]: 'Specify `cause` option to Error\'s constructor.',
+};
+
+const catchClauseSelector = matches([
+	[
+		'CatchClause',
+		':has(ThrowStatement)',
+	].join(''),
+]);
+
+const promiseCatchSelector = matches([
+	[
+		methodCallSelector({method: 'catch', argumentsLength: 1}),
+		':has(ThrowStatement)',
+	].join(''),
+]);
```

Comment:
It seems you didn't cover the second argument in `.then()`.

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/59218e3cd653fdf9ad7e7f5d2666e8c2fa775601/rules/catch-error-name.js#L24-L27